tests/test_hf_api.py

```diff
@@ -1083,6 +1109,30 @@ def test_dataset_info(self):
         self.assertIsInstance(dataset, DatasetInfo)
         self.assertEqual(dataset.sha, DUMMY_DATASET_ID_REVISION_ONE_SPECIFIC_COMMIT)
 
+    @with_production_testing
+    def test_dataset_info_with_file_metadata(self):
+        _api = HfApi()
+        dataset = _api.dataset_info(
+            repo_id=SAMPLE_DATASET_IDENTIFIER,
+            files_metadata=True,
+        )
+        files = dataset.siblings
+        assert files is not None
+        self.assertListEqual(
+            [isinstance(file.blob_id, str) for file in files], [True] * len(files)
+        )
+        self.assertListEqual(
+            [isinstance(file.size, int) for file in files], [True] * len(files)
+        )
+        self.assertTrue(any([file.lfs is not None for file in files]))
+        self.assertListEqual(
+            [
+                file.lfs is None or isinstance(file.lfs, dict) and "sha256" in file.lfs
+                for file in files
+            ],
+            [True] * len(files),
+        )
+
```

Comment:
Other suggestion, I would also refactor the siblings check in a separate helper function and use it in both model and dataset tests. Next time new metadata is added, only this helper would need an update.

EDIT: I realise I am sometimes too much into refactoring. Having self-contained and readable tests is nice as well :)

```suggestion
        self._check_siblings_metadata(files)

    def _check_siblings_metadata(self, files: List[RepoFile])
        """Check requested metadata has been received from the server."""
        at_least_one_lfs = False
        for file in files:
            self.assertTrue(isinstance(file.blob_id, str))
            self.assertTrue(isinstance(file.size, int))
            if file.lfs is not None:
                at_least_one_lfs = True
                self.assertTrue(isinstance(file.lfs, dict) and "sha256" in file.lfs)
        self.assertTrue(at_least_one_lfs)

```